libs/template/class/src/lib/class.directive.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  IterableChanges,
+  IterableDiffer,
+  IterableDiffers,
+  KeyValueChanges,
+  KeyValueDiffer,
+  KeyValueDiffers,
+  OnDestroy,
+  OnInit,
+  Output,
+  Renderer2,
+  ɵisListLikeIterable as isListLikeIterable
+} from '@angular/core';
+import { coerceAllFactory } from '@rx-angular/cdk/coercing';
+import { RxStrategyNames, RxStrategyProvider } from '@rx-angular/cdk/render-strategies';
+import { accumulateObservables } from '@rx-angular/cdk/state';
+import {
+  BehaviorSubject, defer,
+  isObservable,
+  Observable,
+  ObservableInput,
+  ReplaySubject,
+  Subject,
+  Subscription
+} from 'rxjs';
+import { map, switchAll, switchMap, withLatestFrom } from 'rxjs/operators';
+
+type NgClassValues =
+  | string[]
+  | Set<string>
+  | Record<string, boolean>
+  | string
+  | null
+  | undefined;
+
+type RxClassInput =
+  | Observable<NgClassValues>
+  | Record<string, Observable<boolean>>;
+
+@Directive({ selector: '[rxClass]' })
+export class ClassDirective implements OnInit, OnDestroy {
+  private iterableDiffer: IterableDiffer<string> | null = null;
+  private keyValueDiffer: KeyValueDiffer<string, any> | null = null;
+  private rawClass: NgClassValues = null;
+  private newInput = false;
+  private sub = new Subscription();
+  private classValueChanged = false;
+
+  constructor(
+    private readonly iterableDiffers: IterableDiffers,
+    private readonly keyValueDiffers: KeyValueDiffers,
+    private readonly ngEl: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly strategyProvider: RxStrategyProvider
+  ) {}
+
+  private rxClass = coerceAllFactory<NgClassValues>(
+    () => new ReplaySubject(1),
+    switchAll()
+  );
+
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input('rxClass')
+  set rxClass$(value$: RxClassInput) {
+    this.newInput = true;
+    this.rxClass.next(
+      (isObservable(value$) ? value$ : accumulateObservables(value$)).pipe(
+        map((value) => (typeof value === 'string' ? [value] : value))
+      )
```

Comment:
why do we use `accumulateObservables` on a non-observable value?

Reply:
It's actually `Record<string, Observable<boolean>>`.